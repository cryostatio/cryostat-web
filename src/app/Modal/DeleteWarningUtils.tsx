/*
 * Copyright The Cryostat Authors
 *
 * The Universal Permissive License (UPL), Version 1.0
 *
 * Subject to the condition set forth below, permission is hereby granted to any
 * person obtaining a copy of this software, associated documentation and/or data
 * (collectively the "Software"), free of charge and under any and all copyright
 * rights in the Software, and any and all patent rights owned or freely
 * licensable by each licensor hereunder covering either (i) the unmodified
 * Software as contributed to or provided by such licensor, or (ii) the Larger
 * Works (as defined below), to deal in both
 *
 * (a) the Software, and
 * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
 * one is included with the Software (each a "Larger Work" to which the Software
 * is contributed by such licensors),
 *
 * without restriction, including without limitation the rights to copy, create
 * derivative works of, display, perform, and distribute the Software and make,
 * use, sell, offer for sale, import, export, have made, and have sold the
 * Software and the Larger Work(s), and to sublicense the foregoing rights on
 * either these or other terms.
 *
 * This license is subject to the following condition:
 * The above copyright notice and either this complete permission notice or at
 * a minimum a reference to the UPL must be included in all copies or
 * substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
export enum DeleteWarningType {
  DeleteActiveRecordings = 'DeleteActiveRecordings',
  DeleteArchivedRecordings = 'DeleteArchivedRecordings',
  DeleteAutomatedRules = 'DeleteAutomatedRules',
  DeleteEventTemplates = 'DeleteEventTemplates',
  DeleteProbeTemplates = 'DeleteProbeTemplates',
  DeleteActiveProbes = 'DeleteActiveProbes',
  DeleteJMXCredentials = 'DeleteJMXCredentials',
  DeleteCustomTargets = 'DeleteCustomTargets',
}

export interface DeleteWarning {
  id: DeleteWarningType;
  title: string;
  label: string;
  description: string;
  ariaLabel: string;
}

export const DeleteActiveRecordings: DeleteWarning = {
  id: DeleteWarningType.DeleteActiveRecordings,
  title: 'Permanently delete Active Recording?',
  label: 'Delete Active Recording',
  description: `Recording and report data will be lost.`,
  ariaLabel: 'Recording delete warning',
};

export const DeleteArchivedRecordings: DeleteWarning = {
  id: DeleteWarningType.DeleteArchivedRecordings,
  title: 'Permanently delete Archived Recording?',
  label: 'Delete Archived Recording',
  description: `Recording and report data will be lost.`,
  ariaLabel: 'Recording delete warning',
};

export const DeleteAutomatedRules: DeleteWarning = {
  id: DeleteWarningType.DeleteAutomatedRules,
  title: 'Permanently delete Automated Rule?',
  label: 'Delete Automated Rule',
  description: `Rule data will be lost.`,
  ariaLabel: 'Automated rule delete warning',
};

export const DeleteEventTemplates: DeleteWarning = {
  id: DeleteWarningType.DeleteEventTemplates,
  title: 'Permanently delete Event Template?',
  label: 'Delete Event Template',
  description: `Custom event template data will be lost.`,
  ariaLabel: 'Event template delete warning',
};

export const DeleteProbeTemplates: DeleteWarning = {
  id: DeleteWarningType.DeleteProbeTemplates,
  title: 'Permanently delete Probe Template?',
  label: 'Delete Probe Template',
  description: `Custom probe template data will be lost.`,
  ariaLabel: 'Probe template delete warning',
};

export const DeleteActiveProbes: DeleteWarning = {
  id: DeleteWarningType.DeleteActiveProbes,
  title: 'Permanently remove Active Probes from the target?',
  label: 'Remove Active Probes',
  description: `Active probes will be removed from the target.`,
  ariaLabel: 'Active Probes remove warning',
};

export const DeleteJMXCredentials: DeleteWarning = {
  id: DeleteWarningType.DeleteJMXCredentials,
  title: 'Permanently delete JMX Credentials?',
  label: 'Delete JMX Credentials',
  description: `Credential data for this target will be lost.`,
  ariaLabel: 'JMX Credentials delete warning',
};

export const DeleteCustomTargets: DeleteWarning = {
  id: DeleteWarningType.DeleteCustomTargets,
  title: 'Permanently delete Custom Target?',
  label: 'Delete Custom Targets',
  description: `Custom target information will be lost.`,
  ariaLabel: 'Custom Targets delete warning',
};

export const DeleteWarningKinds: DeleteWarning[] = [
  DeleteActiveRecordings,
  DeleteArchivedRecordings,
  DeleteAutomatedRules,
  DeleteEventTemplates,
  DeleteProbeTemplates,
  DeleteActiveProbes,
  DeleteJMXCredentials,
  DeleteCustomTargets,
];

export const getFromWarningMap = (warning: DeleteWarningType): DeleteWarning | undefined => {
  const wt = DeleteWarningKinds.find((t) => t.id === warning);
  return wt;
};
