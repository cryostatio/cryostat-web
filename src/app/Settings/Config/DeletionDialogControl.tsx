/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { DeleteOrDisableWarningType } from '@app/Modal/types';
import { getFromWarningMap } from '@app/Modal/utils';
import { ServiceContext } from '@app/Shared/Services/Services';
import {
  ExpandableSection,
  FormGroup,
  HelperText,
  HelperTextItem,
  Stack,
  StackItem,
  Switch,
} from '@patternfly/react-core';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { SettingTab, UserSetting } from '../types';

const Component = () => {
  const { t } = useTranslation();
  const context = React.useContext(ServiceContext);
  const [state, setState] = React.useState(context.settings.deletionDialogsEnabled());
  const [expanded, setExpanded] = React.useState(false);

  const handleCheckboxChange = React.useCallback(
    (checked, element) => {
      state.set(DeleteOrDisableWarningType[element.target.id], checked);
      context.settings.setDeletionDialogsEnabled(state);
      setState(new Map(state));
    },
    [state, setState, context.settings],
  );

  const handleCheckAll = React.useCallback(
    (checked) => {
      const newState = new Map();
      Array.from(state.entries()).forEach((v) => newState.set(v[0], checked));
      context.settings.setDeletionDialogsEnabled(newState);
      setState(newState);
    },
    [context.settings, setState, state],
  );

  const allChecked = React.useMemo(() => {
    return Array.from(state.entries())
      .map((e) => e[1])
      .reduce((a, b) => a && b);
  }, [state]);

  const switches = React.useMemo(() => {
    return Array.from(state.entries(), ([key, value]) => (
      <StackItem key={key}>
        <Switch
          id={key}
          label={getFromWarningMap(key)?.label || key.toString()}
          isChecked={value}
          onChange={handleCheckboxChange}
        />
      </StackItem>
    ));
  }, [handleCheckboxChange, state]);

  return (
    <>
      <Stack hasGutter>
        <StackItem key="all-deletion-warnings">
          <FormGroup>
            <HelperText>
              <HelperTextItem>{t('SETTINGS.DELETION_DIALOG_CONTROL.SWITCH_DESCRIPTION')}</HelperTextItem>
            </HelperText>
            <Switch
              id="all-deletion-warnings"
              label={t('SETTINGS.DELETION_DIALOG_CONTROL.SWITCH_LABEL')}
              isChecked={allChecked}
              onChange={handleCheckAll}
            />
          </FormGroup>
        </StackItem>
        <StackItem key={'expandable-delete-warning-switch-list'}>
          <ExpandableSection
            toggleText={(expanded ? t('SHOW_LESS', { ns: 'common' }) : t('SHOW_MORE', { ns: 'common' })) || ''}
            onToggle={(_, expanded: boolean) => setExpanded(expanded)}
            isExpanded={expanded}
          >
            {switches}
          </ExpandableSection>
        </StackItem>
      </Stack>
    </>
  );
};

export const DeletionDialogControl: UserSetting = {
  titleKey: 'SETTINGS.DELETION_DIALOG_CONTROL.TITLE',
  descConstruct: 'SETTINGS.DELETION_DIALOG_CONTROL.DESCRIPTION',
  content: Component,
  category: SettingTab.NOTIFICATION_MESSAGE,
};
