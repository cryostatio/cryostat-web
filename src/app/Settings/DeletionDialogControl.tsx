/*
 * Copyright The Cryostat Authors
 *
 * The Universal Permissive License (UPL), Version 1.0
 *
 * Subject to the condition set forth below, permission is hereby granted to any
 * person obtaining a copy of this software, associated documentation and/or data
 * (collectively the "Software"), free of charge and under any and all copyright
 * rights in the Software, and any and all patent rights owned or freely
 * licensable by each licensor hereunder covering either (i) the unmodified
 * Software as contributed to or provided by such licensor, or (ii) the Larger
 * Works (as defined below), to deal in both
 *
 * (a) the Software, and
 * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
 * one is included with the Software (each a "Larger Work" to which the Software
 * is contributed by such licensors),
 *
 * without restriction, including without limitation the rights to copy, create
 * derivative works of, display, perform, and distribute the Software and make,
 * use, sell, offer for sale, import, export, have made, and have sold the
 * Software and the Larger Work(s), and to sublicense the foregoing rights on
 * either these or other terms.
 *
 * This license is subject to the following condition:
 * The above copyright notice and either this complete permission notice or at
 * a minimum a reference to the UPL must be included in all copies or
 * substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

import * as React from 'react';
import { Divider, ExpandableSection, Switch, Stack, StackItem } from '@patternfly/react-core';
import { ServiceContext } from '@app/Shared/Services/Services';
import { UserSetting } from './Settings';
import { DeleteWarningType, getFromWarningMap } from '@app/Modal/DeleteWarningUtils';

const Component = () => {
  const context = React.useContext(ServiceContext);
  const [state, setState] = React.useState(context.settings.deletionDialogsEnabled());
  const [expanded, setExpanded] = React.useState(false);

  const handleCheckboxChange = React.useCallback(
    (checked, element) => {
      state.set(DeleteWarningType[element.target.id], checked);
      context.settings.setDeletionDialogsEnabled(state);
      setState(new Map(state));
    },
    [state, setState, context.settings]
  );

  const handleCheckAll = React.useCallback(
    (checked) => {
      const newState = new Map();
      Array.from(state.entries()).forEach((v) => newState.set(v[0], checked));
      context.settings.setDeletionDialogsEnabled(newState);
      setState(newState);
    },
    [state, setState]
  );

  const allChecked = React.useMemo(() => {
    return Array.from(state.entries())
      .map((e) => e[1])
      .reduce((a, b) => a && b);
  }, [state]);

  const switches = React.useMemo(() => {
    return Array.from(state.entries(), ([key, value]) => (
      <StackItem key={key}>
        <Switch
          id={key}
          label={getFromWarningMap(key)?.label || key.toString()}
          isChecked={value}
          onChange={handleCheckboxChange}
        />
      </StackItem>
    ));
  }, [state]);

  return (
    <>
      <Stack hasGutter>
        <StackItem key="all-deletion-warnings">
          <Switch
            id="all-deletion-warnings"
            label="All Deletion Warnings"
            isChecked={allChecked}
            onChange={handleCheckAll}
          />
        </StackItem>
        <Divider />
        <ExpandableSection
          toggleText={expanded ? 'Show less' : 'Show more'}
          onToggle={setExpanded}
          isExpanded={expanded}
        >
          {switches}
        </ExpandableSection>
      </Stack>
    </>
  );
};

export const DeletionDialogControl: UserSetting = {
  title: 'Show Deletion Dialogs',
  description: 'Enable or disable deletion dialogs by deletion type.',
  content: Component,
};
