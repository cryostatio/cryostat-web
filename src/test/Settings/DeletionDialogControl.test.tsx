/*
 * Copyright The Cryostat Authors
 *
 * The Universal Permissive License (UPL), Version 1.0
 *
 * Subject to the condition set forth below, permission is hereby granted to any
 * person obtaining a copy of this software, associated documentation and/or data
 * (collectively the "Software"), free of charge and under any and all copyright
 * rights in the Software, and any and all patent rights owned or freely
 * licensable by each licensor hereunder covering either (i) the unmodified
 * Software as contributed to or provided by such licensor, or (ii) the Larger
 * Works (as defined below), to deal in both
 *
 * (a) the Software, and
 * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
 * one is included with the Software (each a "Larger Work" to which the Software
 * is contributed by such licensors),
 *
 * without restriction, including without limitation the rights to copy, create
 * derivative works of, display, perform, and distribute the Software and make,
 * use, sell, offer for sale, import, export, have made, and have sold the
 * Software and the Larger Work(s), and to sublicense the foregoing rights on
 * either these or other terms.
 *
 * This license is subject to the following condition:
 * The above copyright notice and either this complete permission notice or at
 * a minimum a reference to the UPL must be included in all copies or
 * substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

import { DeleteOrDisableWarningType } from '@app/Modal/DeleteWarningUtils';
import { DeletionDialogControl } from '@app/Settings/DeletionDialogControl';
import { defaultServices, ServiceContext } from '@app/Shared/Services/Services';
import { cleanup, screen } from '@testing-library/react';
import * as React from 'react';
import renderer, { act } from 'react-test-renderer';
import { renderWithServiceContext, testT } from '../Common';

const defaults = new Map<DeleteOrDisableWarningType, boolean>();
for (const cat in DeleteOrDisableWarningType) {
  defaults.set(DeleteOrDisableWarningType[cat], true);
}

jest.spyOn(defaultServices.settings, 'deletionDialogsEnabled').mockReturnValue(defaults);

describe('<DeletionDialogControl/>', () => {
  beforeEach(() => {
    jest.mocked(defaultServices.settings.setDeletionDialogsEnabled).mockClear();
  });

  afterEach(cleanup);

  it('renders correctly', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(
        <ServiceContext.Provider value={defaultServices}>
          {React.createElement(DeletionDialogControl.content, null)}
        </ServiceContext.Provider>
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should default to enable all deletion dialog', async () => {
    renderWithServiceContext(React.createElement(DeletionDialogControl.content, null));

    const enableSwitch = screen.getByLabelText(testT('SETTINGS.DELETION_DIALOG_CONTROL.SWITCH_LABEL'));
    expect(enableSwitch).toBeInTheDocument();
    expect(enableSwitch).toBeVisible();
    expect(enableSwitch).toBeChecked();
  });

  it('should disable all deletion dialog if switch is off', async () => {
    const { user } = renderWithServiceContext(React.createElement(DeletionDialogControl.content, null));

    const enableSwitch = screen.getByLabelText(testT('SETTINGS.DELETION_DIALOG_CONTROL.SWITCH_LABEL'));
    expect(enableSwitch).toBeInTheDocument();
    expect(enableSwitch).toBeVisible();
    expect(enableSwitch).toBeChecked();

    await user.click(enableSwitch);

    expect(enableSwitch).not.toBeChecked();

    expect(defaultServices.settings.setDeletionDialogsEnabled).toHaveBeenCalledTimes(1);

    const expectedParams = new Map();
    defaults.forEach((_, k) => expectedParams.set(k, false));
    expect(defaultServices.settings.setDeletionDialogsEnabled).toHaveBeenCalledWith(expectedParams);
  });

  it('should turn off switch if any child switch is turned off', async () => {
    const { user } = renderWithServiceContext(React.createElement(DeletionDialogControl.content, null));

    const enableSwitch = screen.getByLabelText(testT('SETTINGS.DELETION_DIALOG_CONTROL.SWITCH_LABEL'));
    expect(enableSwitch).toBeInTheDocument();
    expect(enableSwitch).toBeVisible();
    expect(enableSwitch).toBeChecked();

    const expandButton = screen.getByText(testT('SHOW_MORE', 'common'));
    expect(expandButton).toBeInTheDocument();
    expect(expandButton).toBeVisible();

    await user.click(expandButton);

    const activeRecordingSwitch = screen.getByLabelText('Delete Active Recording');
    expect(activeRecordingSwitch).toBeInTheDocument();
    expect(activeRecordingSwitch).toBeVisible();
    expect(activeRecordingSwitch).toBeChecked();

    await user.click(activeRecordingSwitch);

    expect(activeRecordingSwitch).not.toBeChecked();
    expect(enableSwitch).not.toBeChecked();
  });
});
