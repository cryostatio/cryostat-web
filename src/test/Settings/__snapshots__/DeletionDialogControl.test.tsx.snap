// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeletionDialogControl/> renders correctly 1`] = `
<div
  className="pf-v5-l-stack pf-m-gutter"
>
  <div
    className="pf-v5-l-stack__item"
  >
    <div
      className="pf-v5-c-form__group"
    >
      <div
        className="pf-v5-c-form__group-control"
      >
        <div
          className="pf-v5-c-helper-text"
        >
          <div
            className="pf-v5-c-helper-text__item"
          >
            <span
              className="pf-v5-c-helper-text__item-text"
            >
              Enable or disable deletion dialogs by deletion type.
            </span>
          </div>
        </div>
        <label
          className="pf-v5-c-switch"
          data-ouia-component-id="OUIA-Generated-Switch-1"
          data-ouia-component-type="PF5/Switch"
          data-ouia-safe={true}
          htmlFor="all-deletion-warnings"
        >
          <input
            aria-label=""
            aria-labelledby="all-deletion-warnings-on"
            checked={true}
            className="pf-v5-c-switch__input"
            disabled={false}
            id="all-deletion-warnings"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="pf-v5-c-switch__toggle"
          />
          <span
            aria-hidden="true"
            className="pf-v5-c-switch__label pf-m-on"
            id="all-deletion-warnings-on"
          >
            All deletion warnings
          </span>
          <span
            aria-hidden="true"
            className="pf-v5-c-switch__label pf-m-off"
            id="all-deletion-warnings-off"
          >
            All deletion warnings
          </span>
        </label>
      </div>
    </div>
  </div>
  <div
    className="pf-v5-l-stack__item"
  >
    <div
      className="pf-v5-c-expandable-section"
    >
      <button
        aria-controls="expandable-section-content-17177484467765uqbuek5d18"
        aria-expanded={false}
        className="pf-v5-c-expandable-section__toggle"
        id="expandable-section-toggle-1717748446776vbr16rjdf6l"
        onClick={[Function]}
        type="button"
      >
        <span
          className="pf-v5-c-expandable-section__toggle-icon"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            className="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
        <span
          className="pf-v5-c-expandable-section__toggle-text"
        >
          Show more
        </span>
      </button>
      <div
        aria-labelledby="expandable-section-toggle-1717748446776vbr16rjdf6l"
        className="pf-v5-c-expandable-section__content"
        hidden={true}
        id="expandable-section-content-17177484467765uqbuek5d18"
        role="region"
      >
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-2"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DeleteActiveRecordings"
          >
            <input
              aria-label=""
              aria-labelledby="DeleteActiveRecordings-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DeleteActiveRecordings"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DeleteActiveRecordings-on"
            >
              Delete Active Recording
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DeleteActiveRecordings-off"
            >
              Delete Active Recording
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-3"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DeleteArchivedRecordings"
          >
            <input
              aria-label=""
              aria-labelledby="DeleteArchivedRecordings-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DeleteArchivedRecordings"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DeleteArchivedRecordings-on"
            >
              Delete Archived Recording
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DeleteArchivedRecordings-off"
            >
              Delete Archived Recording
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-4"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DeleteAutomatedRules"
          >
            <input
              aria-label=""
              aria-labelledby="DeleteAutomatedRules-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DeleteAutomatedRules"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DeleteAutomatedRules-on"
            >
              Delete Automated Rule
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DeleteAutomatedRules-off"
            >
              Delete Automated Rule
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-5"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DisableAutomatedRules"
          >
            <input
              aria-label=""
              aria-labelledby="DisableAutomatedRules-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DisableAutomatedRules"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DisableAutomatedRules-on"
            >
              Disable Automated Rule
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DisableAutomatedRules-off"
            >
              Disable Automated Rule
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-6"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DeleteEventTemplates"
          >
            <input
              aria-label=""
              aria-labelledby="DeleteEventTemplates-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DeleteEventTemplates"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DeleteEventTemplates-on"
            >
              Delete Event Template
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DeleteEventTemplates-off"
            >
              Delete Event Template
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-7"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DeleteProbeTemplates"
          >
            <input
              aria-label=""
              aria-labelledby="DeleteProbeTemplates-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DeleteProbeTemplates"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DeleteProbeTemplates-on"
            >
              Delete Probe Template
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DeleteProbeTemplates-off"
            >
              Delete Probe Template
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-8"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DeleteActiveProbes"
          >
            <input
              aria-label=""
              aria-labelledby="DeleteActiveProbes-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DeleteActiveProbes"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DeleteActiveProbes-on"
            >
              Remove active probes
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DeleteActiveProbes-off"
            >
              Remove active probes
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-9"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DeleteCredentials"
          >
            <input
              aria-label=""
              aria-labelledby="DeleteCredentials-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DeleteCredentials"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DeleteCredentials-on"
            >
              Delete Credentials
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DeleteCredentials-off"
            >
              Delete Credentials
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-10"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DeleteCustomTargets"
          >
            <input
              aria-label=""
              aria-labelledby="DeleteCustomTargets-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DeleteCustomTargets"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DeleteCustomTargets-on"
            >
              Delete Custom Targets
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DeleteCustomTargets-off"
            >
              Delete Custom Targets
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-11"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DeleteDashboardLayout"
          >
            <input
              aria-label=""
              aria-labelledby="DeleteDashboardLayout-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DeleteDashboardLayout"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DeleteDashboardLayout-on"
            >
              Delete Dashboard layouts
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DeleteDashboardLayout-off"
            >
              Delete Dashboard layouts
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-12"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="DeleteLayoutTemplate"
          >
            <input
              aria-label=""
              aria-labelledby="DeleteLayoutTemplate-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="DeleteLayoutTemplate"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="DeleteLayoutTemplate-on"
            >
              Delete Layout Template
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="DeleteLayoutTemplate-off"
            >
              Delete Layout Template
            </span>
          </label>
        </div>
        <div
          className="pf-v5-l-stack__item"
        >
          <label
            className="pf-v5-c-switch"
            data-ouia-component-id="OUIA-Generated-Switch-13"
            data-ouia-component-type="PF5/Switch"
            data-ouia-safe={true}
            htmlFor="ClearDashboardLayout"
          >
            <input
              aria-label=""
              aria-labelledby="ClearDashboardLayout-on"
              checked={true}
              className="pf-v5-c-switch__input"
              disabled={false}
              id="ClearDashboardLayout"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="pf-v5-c-switch__toggle"
            />
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-on"
              id="ClearDashboardLayout-on"
            >
              Clear Dashboard Layout
            </span>
            <span
              aria-hidden="true"
              className="pf-v5-c-switch__label pf-m-off"
              id="ClearDashboardLayout-off"
            >
              Clear Dashboard Layout
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
